src/components/dashboard/DashboardDetail.js

```diff
@@ -133,60 +76,62 @@ class DashboardDetail extends Component {
-               to='/authorized-apps'
-               accessKeys={authorizedApps}
-            />
-            <DashboardKeys
-               image={AuthorizedGreyImage}
-               title='Full Access Keys'
-               to='/full-access-keys'
-               accessKeys={fullAccessKeys}
-            />
-            {false ? (
-               <DashboardSection
-                  notice={notice}
-                  handleNotice={this.handleNotice}
-               >
-                  <DashboardActivity
-                     loader={loader}
-                     image={activityGreyImage}
-                     title='Activity'
-                     to='/'
-                     activity={activity}
-                  />
-                  <DashboardActivity
-                     loader={loader}
-                     image={AuthorizedGreyImage}
-                     title='Authorized Apps'
-                     to='/authorized-apps'
-                     activity={authorizedApps}
-                  />
-                  <DashboardActivity
-                     loader={loader}
-                     image={ContactsGreyImage}
-                     title='New Contacts'
-                     to='/contacts'
-                     activity={newcontacts}
-                  />
-               </DashboardSection>
-            ) : null}
+            <DashboardSection
+               notice={notice}
+               handleNotice={this.handleNotice}
+            >
+               <DashboardActivity
+                  loader={loader}
+                  image={activityGreyImage}
+                  title='Activity'
+                  to={`${process.env.EXPLORER_URL || 'https://explorer.nearprotocol.com'}/accounts/${accountId}`}
+                  transactions={transactions}
+               />
+               <DashboardKeys
+                  image={AuthorizedGreyImage}
+                  title='Authorized Apps'
+                  to='/authorized-apps'
+                  accessKeys={authorizedApps}
+               />
+               <DashboardKeys
+                  image={AuthorizedGreyImage}
+                  title='Full Access Keys'
+                  to='/full-access-keys'
+                  accessKeys={fullAccessKeys}
+               />
+            </DashboardSection>
          </PageContainer>
       )
    }
 }
 
 const mapDispatchToProps = {
-   getAccessKeys
+   getAccessKeys,
+   getTransactions
 }
 
-const mapStateToProps = ({ account }) => ({
-   ...account,
-   authorizedApps: account.authorizedApps,
-   fullAccessKeys: account.fullAccessKeys
-})
+const mapStateToProps = ({ account }) => {
+   const transactions = account.transactions 
+      ? account.transactions.reduce((x, t) => {
+            try {
+               t.actions = JSON.parse(t.actions)
+            } catch {}
+            return x.concat(t.actions.map((a) => {
+               return {
+                  ...t,
+                  action: typeof a == 'object' ? [a] : [{[a]: {}}]
```

Comment:
not sure why this is needed, maybe needs some explanation in comment

Reply:
@vgrichina Oh, I was also confused. The reason for this thing is that nearcore serializes a list of actions as a list of (object | string) (serde serializes empty struct as a string as opposed to an object):

```
[
  "CreateAccount",
  {"Transfer": {"deposit":"10000000000000000000"}},
  {"AddKey":{"access_key":{"nonce":0,"permission":"FullAccess"},"public_key":"ed25519:4x9s9hpk5ipYJufQXPFEPxrYjW9VStLJtfpRmETuwVHG"}}
]
```

This line translates `"CreateAccount"` into `{"CreateAccount": {}}`.

I hope to address this on the nearcore side one day...

Reply:
yes, it's necessary for now.

Reply:
@marcinbodnar just add a comment, as it's not immediately obvious what's going on. Or maybe even better – extract it to some function, e.g. `postprocessSerdeStruct` which would have this explanation.